src/hooks/useOnboardingForm.tsx

```diff
@@ -1,7 +1,12 @@
 "use client";
-import { ReactElement, useState } from "react";
+import { FC, useState } from "react";
 
-export function useOnboardingForm(initialStep = 0, steps: ReactElement[]) {
+export type FormProps = {
+  onSubmit: () => void;
+  moveNextStep: () => void;
+};
+
+export function useOnboardingForm(initialStep = 0, steps: FC<FormProps>[]) {
```

Comment:
To follow-up on the idea of relating each step component to its validation schema, we can change this `steps` to have two things

```suggestion

type Step = {
  component: FC<FormProps>;
  schema: ValidationSchemaType; // not sure if this type exists, probably not but I think you got the idea
}

export function useOnboardingForm(initialStep = 0, steps: Step[]) {
```

This way we can move the validation step to this hook, and leave it responsible to handle the form logic

Reply:
Ok!

Reply:
As we agreed in the daily, we'll do this in a future PR